src/repart/repart.c

```diff
@@ -5054,32 +5069,48 @@ static int partition_format_verity_sig(Context *context, Partition *p) {
         if (PARTITION_EXISTS(p))
                 return 0;
 
-        if (!context->private_key)
+        assert_se(hp = p->siblings[VERITY_HASH]);
+        assert(!hp->dropped);
+        assert_se(rp = p->siblings[VERITY_DATA]);
+        assert(!rp->dropped);
+
+        if (!sd_id128_is_null(hp->current_uuid) && !sd_id128_is_null(rp->current_uuid)) {
+                char *root_hash_hex = strjoina(SD_ID128_TO_STRING(rp->current_uuid), SD_ID128_TO_STRING(hp->current_uuid));
```

Comment:
This deserves a comment with a link to the spec or something that defines that the UUID of these partitions combined is the roothash.